File: HackerEarth/Rohan and flower.java
 // Rohan and flower*

 // https://www.hackerearth.com/problem/algorithm/rohan-and-flower/


import java.util.*;
class TestClass {
    public static void main(String args[] ) throws Exception {
    
    Scanner s=new Scanner(System.in);
    int l=s.nextInt();
    int h=s.nextInt();
    int a[]=new int[l];
    for(int i=0;i<l;i++)
    a[i]=s.nextInt();
    int c=0;
   for(int i=0;i<l-2;i++)
   {
	for(int j=i+1;j<l-1;j++)
	{
		for(int k=j+1;k<l;k++)
		{
			  int p=a[i]*a[j]*a[k];
				if(p%h==0)
					c=c+1;
		}
	}
   }
        System.out.println(c);
    }
}

File: HackerEarth/Rohan and flower.py
 # // Rohan and flower*

 # // https://www.hackerearth.com/problem/algorithm/rohan-and-flower/



f=int(raw_input())
h=int(raw_input())
a=[]
a=map(int,raw_input().split())
a.sort()
l=len(a)
c=0
for i in range(0,l-2):
	for j in range(i+1,l-1):
		for k in range(j+1,l):
				p=int(a[i]*a[j]*a[k])
				if p%h==0:
					c=c+1
print c